redis.go

```diff
@@ -412,6 +412,20 @@ func (c *baseClient) process(ctx context.Context, cmd Cmder) error {
 	return lastErr
 }
 
+func (c *baseClient) isProblematicMethodsOfSearchResp3(ctx context.Context, cmd Cmder) bool {
+	if c.opt.Protocol != 3 {
+		return false
+	}
+
+	switch cmd.(type) {
+	case *AggregateCmd, *FTInfoCmd, *FTSpellCheckCmd, *FTSearchCmd, *FTSynDumpCmd:
+		internal.Logger.Printf(ctx, "Some RESP3 results for Redis Query Engine responses may change. Refer to the readme for guidance")
```

Comment:
The logic should be different here. So the algorithm is:

1. If `protocol == 2` - we allow to use Search module without limitations.
2. If `protocol == 3 && UnstableResp3SearchModule !== True` - we throw this exception "Some RESP3 results for Redis Query Engine responses may change. Refer to the readme for guidance"
3. If `protocol == 3 && UnstableResp3SearchModule == True` - we allows to use Search module with the limitation that command replies will be raw.